src/snowflake/snowpark/mock/_functions.py

```diff
@@ -308,6 +336,63 @@ def mock_to_date(
     )
 
 
+@patch("as_timestamp_ntz")
+def mock_as_timestamp_ntz(expr1: ColumnEmulator):
+    # NTZ timestamps can be recognized by the lack of tzinfo
+    filtered = [
+        x if isinstance(x, datetime.datetime) and x.tzinfo is None else None
+        for x in expr1
+    ]
+    return ColumnEmulator(
+        data=filtered,
+        sf_type=ColumnType(
+            TimestampType(TimestampTimeZone.NTZ), expr1.sf_type.nullable
+        ),
+        dtype=object,
+    )
+
+
+@patch("as_timestamp_ltz")
+def mock_as_timestamp_ltz(expr1: ColumnEmulator):
+    # LTZ timestamp can be recognized by checking the utc offset for the timestamp is the same as the utcoffset for a local time
+    local_offset = LocalTimezone.to_local_timezone(datetime.datetime.now()).utcoffset()
+    filtered = [
+        x
+        if isinstance(x, datetime.datetime) and x.utcoffset() == local_offset
+        else None
+        for x in expr1
+    ]
+    return ColumnEmulator(
+        data=filtered,
+        sf_type=ColumnType(
+            TimestampType(TimestampTimeZone.NTZ), expr1.sf_type.nullable
+        ),
+        dtype=object,
+    )
+
+
+@patch("as_timestamp_tz")
+def mock_as_timestamp_tz(expr1: ColumnEmulator):
+    # TZ timestamps appear to be timestamps that have tzinfo, but it isn't the local tzinfo
+    # This logic seems incorrect, but it matches what the non-local version does
```

Comment:
This logic matches what I observed when running manual queries as well as what the non-local testing tests return. It doesn't seem particularly consistent so if anyone has a link to some documentation that better defines this operation I'd be interested in seeing it.

Reply:
In order to unblock merging this change I've decided to move the as_timestamp mocks out into a separate PR.